### ISSUE TITLE
while retiring/landing a worker, add an option to block until it's landed/retired

### ISSUE THREAD

**Issue by kalfa** [concourse/fly#231]
Fly returns immediately after the retire/land API call has been made to TSA.

The use case is a provisioner which is called during resource destruction (e.g. terraform provisioner with when="destroy"), which should know when TSA considers the worker which is being destroyed officially "retired", so it can do other operations (i.e. shut down the machine).

desirable result: have a simple way to understand when the machine is ready to go offline

**Comment by jama-pivotal** [comment]
Somewhat related to https://github.com/concourse/concourse/issues/1458

**Comment by jama-pivotal** [comment]
Somewhat related to https://github.com/concourse/concourse/issues/1458

**Comment by kalfa** [comment]
related in a cause/effect way: this ticket not implemented can lead to "unknown handle" or similar errors after worker's destruction (although it's not the only cause).

**Comment by jama-pivotal** [comment]
Agreed @kalfa , only linking because it helps with our visual clustering on our projects view: https://project.concourse-ci.org/projects/Runtime 😄
